src/main/java/build/buf/protovalidate/Value.java

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package build.buf.protovalidate.internal.evaluator;
```

Comment:
Unclear if this type really carries its weight as a top-level class in the public API. Maybe the `Violation` wrapper type could just return `Object` for field and rule values, and the Java docs could describe all of the possible types it could return? I assume that would be limited to the three composite types, `Message` (and concrete sub-types), `List<Value>`, and `Map<Value,Value>`, plus `String`, `byte[]`, and boxed primitives for the numeric/bool types?

Reply:
I think if we're going to do that, we should probably directly expose the `FieldDescriptor` like we do in Go. It's not technically needed, but I'm pretty sure it would be useful, and we always have it.

Reply:
Dropped Value from the public API + added corresponding FieldDescriptors to Violation. That said, I don't know what Java protobuf does for all of the dynamic cases, so I haven't actually added docs describing what the objects can be.